javascript/javascript2/week3/fp.js

```diff
@@ -0,0 +1,83 @@
+let input = document.getElementById('myinput');
+let button = document.getElementById('start');
+let rButton = document.getElementById('reset');
+let leftLabel = document.querySelector('.leftsection label');
+let rightLabel = document.querySelector('.rightsection label');
+let lWinMessage = document.querySelector('.leftsection label:last-child');
+let rWinMessage = document.querySelector('.rightsection label:last-child');
+
+let countS = 0;
+let countL = 0;
+
+
+// Will be executed when I press "Start game"
+function getInputValue() {
+    // Only one game can be played at a time(Once the game is started I cannot enter time again and press start game again)
+    input.disabled = true;
+    button.disabled = true;
+    let time = input.value;
+    if (time === 0) {
```

Comment:
time is going to be a string, so you either need to convert it to a number or do a string comparison.

`Number('') === 0` so perhaps converting the time at the start to a number would be best like

```
let time = Number(time)
```

Reply:
Oh I noticed this now. I think I miss these small details .SInce the program was working I didnt think about this. Thanks for the detailed feedback. I have corrected a small logic in the program and I have converted to Number and changed to more meaningful variable names as mentioned above.

Reply:
let input = document.getElementById("myinput");
let button = document.getElementById("start");
let rButton = document.getElementById("reset");
let leftLabel = document.querySelector(".leftsection label");
let rightLabel = document.querySelector(".rightsection label");
let lWinMessage = document.querySelector(".leftsection label:last-child");
let rWinMessage = document.querySelector(".rightsection label:last-child");

let countS = 0;
let countL = 0;

// Will be executed when I press "Start game"
function startGame() {
  // Only one game can be played at a time(Once the game is started I cannot enter time again and press start game again)
  input.disabled = true;
  button.disabled = true;

  let time = input.value;
  time = Number(time);
  
 
  if (time === 0) {
    alert(`Please enter the time to proceed`);
    input.disabled = false;
    button.disabled = false;
  } else {
    //  when ever I start the game I should start with 0
    countS = 0;
    countL = 0;
    leftLabel.innerHTML = countS;
    rightLabel.innerHTML = countL;
    window.addEventListener("keydown", keyClicked); // I want keydown action to work only after the game has started. Thats why its here.
    setTimeout(() => {
      //I am disabling here because I am not able to access these fields after the game has completed(for next iteration).
      input.disabled = false;
      button.disabled = false;
      window.removeEventListener("keydown", keyClicked); // I do not want the keydown to work after the game has finished.
      if (leftLabel.innerHTML === 0 && rightLabel.innerHTML === 0) {
        alert(`Press the keys to win!!`);
      } else {
        console.log("Sorry, Your time is up!!");
        console.log(leftLabel.innerHTML);
        console.log(rightLabel.innerHTML);
      }
      if (countS > countL) {
        lWinMessage.innerHTML = `Congrats,You won!`;
        rWinMessage.innerHTML = `Sorry,You lost!`;
        let confettiSettings = { target: "left" };
        let confetti = new ConfettiGenerator(confettiSettings);
        confetti.render();
      } else if (countS < countL) {
        rWinMessage.innerHTML = `Congrats,You won!`;
        lWinMessage.innerHTML = `Sorry,You lost!`;
        let confettiSettings = { target: "right" };
        let confetti = new ConfettiGenerator(confettiSettings);
        confetti.render();
      } else {
        lWinMessage.innerHTML = `Game Draw!`;
        rWinMessage.innerHTML = `Game Draw!`;
      }
    }, time * 1000);
  }
}
button.addEventListener("click", startGame);

// This function checks for S and L and stores the count of presses.
function keyClicked(event) {
  if (event.code === "KeyS") {
    countS++;
  } else if (event.code === "KeyL") {
    countL++;
  }
  leftLabel.innerHTML = countS;
  rightLabel.innerHTML = countL;
}

// By doing this I can reset all the values in my game as initial.
function restart() {
  window.location.href = "";
}

rButton.addEventListener("click", restart);